### PR TITLE
Update ServerInHandle comments

### LINKED ISSUE #1436: Errors returned from InTapHandle are lost
Please answer these questions before submitting your issue.

### What version of gRPC are you using?

1.5.1

### What version of Go are you using (`go version`)?

1.8.3

### What operating system (Linux, Windows, …) and version?

Ubuntu Linux, 16.04

### What did you do?

Returned a PermissionDenied gRPC error from the server's installed grpc.InTapHandle 

### What did you expect to see?

The returned error sent to the client, or at least an error that might help locate the source of the error.

### What did you see instead?

Client gets sent gibberish:

rpc error: code = Unavailable desc = stream terminated by RST_STREAM with error code: 7

Also, no error is reported on the server either. the Serve() method keeps serving silently.

### PR DESCRIPTION
**Pull request by menghanl**
fixes #1436

### CODE REVIEW

File: tap/tap.go [hunk 1]
```diff
@@ -32,8 +32,19 @@ type Info struct {
...
 }
 
-// ServerInHandle defines the function which runs when a new stream is created
-// on the server side. Note that it is executed in the per-connection I/O goroutine(s) instead
-// of per-RPC goroutine. Therefore, users should NOT have any blocking/time-consuming
-// work in this handle. Otherwise all the RPCs would slow down.
+// ServerInHandle defines the function which runs before a new stream is created
+// on the server side. If it returns a non-nil error, the stream will not be
+// created and a RST_STREAM will be sent back to the client with REFUSED_STREAM.
+// The client will receive a RPC error "code = Unavailable, desc = stream
```
Comment (dfawley) [inline tap/tap.go]: Nit: *an* RPC error

File: tap/tap.go [hunk 2]
```diff
@@ -32,8 +32,19 @@ type Info struct {
...
+// terminated by RST_STREAM with error code: REFUSED_STREAM".
+//
+// It's intended to be used in situations where you don't want to waste the
+// resource to accept the new stream (e.g. rate-limiting). And the content of
```
Comment (dfawley) [inline tap/tap.go]: Nit: resource*s*

File: tap/tap.go [hunk 3]
```diff
@@ -32,8 +32,19 @@ type Info struct {
...
+// the error will be ignored and won't be sent back to the client. For other
+// general usages, please use interceptors.
+//
+// Note that it is executed in the per-connection I/O goroutine(s) instead of
+// per-RPC goroutine. Therefore, users should NOT have any
+// blocking/time-consuming work in this handle. Otherwise all the RPCs would
+// slow down.
```
Comment (dfawley) [inline tap/tap.go]: Should we add a note about how that means it also won't be called multiple times at once, and so doesn't need to be thread-safe?

**Comment by menghanl** [comment]
Thanks for the review. All fixed. PTAL.

**Pull request merged**